src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,42 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (DataProcessingException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movie;
+    }
+    
+    @Override
+    public Optional<Movie> get(Long id) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = sessionFactory.openSession();
+        return Optional.ofNullable(session.get(Movie.class, id));
```

Comment:
should be done with try-with-resources. READ methods can be done with it because you don't use transaction, still session call and methods may cause exception and you need to cover them with try-catch block. Try with resources will also close opened session, which is not closed in this code.